internal/konnect/node_agent.go

```diff
@@ -112,11 +141,22 @@ func (a *NodeAgent) updateNodeLoop() {
 	}
 }
 
+// receiveStatus receives the necessary information to set the status.
+func (a *NodeAgent) receiveStatus() {
+	for {
+		select {
+		case a.hasTranslationFailure = <-a.hasTranslationFailureChan:
+		case a.sendCondifError = <-a.sendConfigErrorChan:
+		}
+	}
```

Comment:
We should pass `context` down from `Run` and allow aborting this loop in case we are shutting down and the context gets cancelled (same for `updateNodeLoop`).

Reply:
Added `context`.